test/correctness/double_buffer.cpp

```diff
@@ -0,0 +1,385 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    if (get_jit_target_from_environment().arch == Target::WebAssembly) {
+        printf("[SKIP] WebAssembly does not support async() yet.\n");
+        return 0;
+    }
+
+    // Basic compute-root async producer
+    {
+        Func producer("producer"), consumer("consumer");
+        Var x, y, xo, yo, xi, yi;
+
+        producer(x, y) = x + y;
+        consumer(x, y) = producer(x - 1, y - 1) + producer(x, y) + producer(x + 1, y + 1);
+
+        consumer
+            .compute_root()
+            .tile(x, y, xo, yo, xi, yi, 16, 16, TailStrategy::RoundUp);
+        producer
```

Comment:
Suggestion: even though this is just a test, it's worth adding some comments about what this schedule is doing and how the storage schedules interact.

Reply:
I've made test descriptions a bit more concrete.